lightning/src/util/test_utils.rs

```diff
@@ -72,17 +73,27 @@ impl chaininterface::FeeEstimator for TestFeeEstimator {
 	}
 }
 
-pub struct TestRouter {}
+pub struct TestRouter<'a> {
+	pub network_graph: Arc<NetworkGraph<&'a TestLogger>>,
+}
+
+impl<'a> TestRouter<'a> {
+	pub fn new(network_graph: Arc<NetworkGraph<&'a TestLogger>>) -> Self {
+		Self { network_graph }
+	}
+}
 
-impl Router for TestRouter {
+impl<'a> Router for TestRouter<'a> {
 	fn find_route(
-		&self, _payer: &PublicKey, _params: &RouteParameters, _first_hops: Option<&[&channelmanager::ChannelDetails]>,
-		_inflight_htlcs: InFlightHtlcs
+		&self, payer: &PublicKey, params: &RouteParameters, first_hops: Option<&[&channelmanager::ChannelDetails]>,
+		inflight_htlcs: InFlightHtlcs
 	) -> Result<Route, msgs::LightningError> {
-		Err(msgs::LightningError {
-			err: String::from("Not implemented"),
-			action: msgs::ErrorAction::IgnoreError
-		})
+		let logger = TestLogger::new();
+		find_route(
+			payer, params, &self.network_graph, first_hops, &logger,
+			&ScorerAccountingForInFlightHtlcs::new(&mut TestScorer::with_penalty(0), inflight_htlcs),
```

Comment:
Looks like this needs to take `TestScorer` by value in this commit rather than the next one?